tests/scalers/beanstalkd/beanstalkd_test.go

```diff
@@ -0,0 +1,227 @@
+    spec:
+      containers:
+      - name: beanstalkd-put-job
+        image: docker.io/sitecrafting/beanstalkd-cli
+        command: ["/bin/sh"]
+        args: ["-c", "for run in $(seq 1 10); do beanstalkd-cli --host=beanstalkd put \"Test Job\"; done;"]
+      restartPolicy: OnFailure
+`
+
+	scaledObjectDelayedTemplate = `
+apiVersion: keda.sh/v1alpha1
+kind: ScaledObject
+metadata:
+  name: {{.ScaledObjectName}}
+  namespace: {{.TestNamespace}}
+spec:
+  scaleTargetRef:
+    name: {{.DeploymentName}}
+  maxReplicaCount: 5
+  minReplicaCount: 1
+  triggers:
+  - type: beanstalkd
+    metadata:
+      server: beanstalkd.{{.TestNamespace}}:11300
+      value: "5"
+      tube: {{.BeanstalkdTubeName}}
+      includeDelayed: "true"
+`
+
+	deploymentTemplate = `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{.DeploymentName}}
+  namespace: {{.TestNamespace}}
+  labels:
+    app: nginx-deployment
+spec:
+  replicas: 0
+  selector:
+    matchLabels:
+      app: nginx-deployment
+  template:
+    metadata:
+      labels:
+        app: nginx-deployment
+    spec:
+      containers:
+      - name: nginx-deployment
+        image: nginxinc/nginx-unprivileged
+        ports:
+        - containerPort: 80
+`
+)
+
+func TestBeanstalkdScaler(t *testing.T) {
+	// setup
+	t.Log("--- setting up ---")
+	// Create kubernetes resources
+	kc := GetKubernetesClient(t)
+	data, templates := getTemplateData()
+	t.Cleanup(func() {
+		DeleteKubernetesResources(t, testNamespace, data, templates)
+	})
+
+	CreateKubernetesResources(t, kc, testNamespace, data, templates)
+
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, beanstalkdDeploymentName, testNamespace, 1, 60, 1),
+		"replica count should be 0 after a minute")
+
+	// Add beanstalkd jobs
+	addBeanstalkdJobs(t, kc, &data)
+
+	// test activation
+	testActivation(t, kc, data)
+
+	// test scaling
+	testScale(t, kc, data)
+}
+
+func getTemplateData() (templateData, []Template) {
+	return templateData{
+			TestNamespace:            testNamespace,
+			ScaledObjectName:         scaledObjectName,
+			DeploymentName:           deploymentName,
+			BeanstalkdDeploymentName: beanstalkdDeploymentName,
+			BeanstalkdTubeName:       beanstalkdTubeName,
+			BeanstalkdPutJobName:     beanstalkdPutJobName,
+		}, []Template{
+			{Name: "beanstalkdDeploymentTemplate", Config: beanstalkdDeploymentTemplate},
+			{Name: "deploymentTemplate", Config: deploymentTemplate},
+		}
+}
+
+func addBeanstalkdJobs(t *testing.T, kc *kubernetes.Clientset, data *templateData) {
+	// run putJob
+	KubectlReplaceWithTemplate(t, data, "beanstalkdPutJobsTemplate", beanstalkdPutJobsTemplate)
+	require.True(t, WaitForJobSuccess(t, kc, beanstalkdPutJobName, testNamespace, 30, 2), "Job should run successfully")
+}
+
+func testActivation(t *testing.T, kc *kubernetes.Clientset, data templateData) {
+	t.Log("--- testing activation---")
+
+	KubectlApplyWithTemplate(t, data, "scaledObjectActivationTemplate", scaledObjectActivationTemplate)
```

Comment:
We only change the ScaledObject when there isn't another option. To test the activation and the scaling, is it possible to add and remove jobs? I mean
- Activation -> enqueue some jobs below the activation value
- Scaling out -> enqueue more jobs over the activation value
- Scaling in -> remove the jobs

Reply:
Thanks for the review. I've updated the testing to use one ScaledObject now which tests activation, scaling in and scaling out.